fabric/src/main/java/dev/architectury/impl/fabric/EventChatDecorator.java

```diff
@@ -19,68 +19,8 @@
 
 package dev.architectury.impl.fabric;
 
-import dev.architectury.event.events.common.ChatEvent;
-import net.minecraft.network.chat.ChatDecorator;
 import net.minecraft.network.chat.Component;
-import net.minecraft.network.chat.MessageSignature;
-import net.minecraft.network.chat.PlayerChatMessage;
-import net.minecraft.server.level.ServerPlayer;
-import net.minecraft.server.network.FilteredText;
-import org.jetbrains.annotations.Nullable;
 
-import java.util.concurrent.CompletableFuture;
-
-public class EventChatDecorator implements ChatDecorator {
+public class EventChatDecorator {
     public static final Component CANCELLING_COMPONENT = Component.literal("THIS SHOULDN'T BE DISPLAYED, ARCHITECTURY SPECIFIC STRING DO NOT IMITATE THIS");
```

Comment:
Is there really no other way to "cancel" the player's chat message than to send a weird special component? For instance, what happens if the Future returns null or completes exceptionally?

Reply:
Not really, this is a compromise for this badly designed event

Reply:
Honestly, yeah, then it's... fine, considering 1.19 has basically been superseded by 1.19.2+ anyways